tasks/image-segmentation/src/validation.py

```diff
@@ -0,0 +1,192 @@
+        sampleData = sample.load()
+
+        if hasDotAnnotation(sampleData.annotation):
+            logging.warning(f">> [Image Segmentation] Sample \"{sample.name}\" (ID: {sample.id}) has invalid annotation (too few coordinates). Skipping Sample")
+            continue
+
+        resized, _, _ = resizeWithPadding(sampleData.image, imageSize, imageSize)
+        normalized = resized / 255
+
+        groundtruth = sampleData.extractSegmentationMask(classes)
+        groundtruth, vPadding, hPadding = resizeWithPadding(groundtruth, imageSize, imageSize)
+
+        testImgs.append(normalized)
+        testMasks.append(groundtruth)
+        testMasksPadding.append((vPadding, hPadding))
+
+    testImages = np.array(testImgs)
+
+    return testImages, testMasks, testMasksPadding
+
+
+def loadModel(modelPath: Path) -> tf.lite.Interpreter:
+    modelInterpreter = tf.lite.Interpreter(str(modelPath / "model.tflite"))
+    return modelInterpreter
+
+
+def validation(taskRun: TaskRun) -> None:
+    dataset = taskRun.dataset
+    imageSize: int = taskRun.parameters["imageSize"]
+    batchSize: int = taskRun.parameters["batchSize"]
+    dataset.classes.exclude(taskRun.parameters["excludedClasses"])
+
+    if taskRun.parameters["trainedModel"] is None:
+        raise RuntimeError("Model id used for image segmentation that needs validation is not valid")
+
+    model: Model = taskRun.parameters["trainedModel"]
+    model.download()
+    modelInterpreter = loadModel(model.path)
+
+    iouScores: list[float] = []
+    fieldNamesSamples = ["Sample ID", "Sample Name", "IoU Score", "Accuracy"]
+    csvSamplesData: list[dict[str, str]] = []
+
+    for startIndex in range(0, dataset.count, batchSize):
+        batchSamples = dataset.samples[startIndex:startIndex + batchSize]
+        testImages, testMasks, testMasksPadding = predictionPrepare(batchSamples, imageSize, dataset.classes)
+
+        prediction = run(modelInterpreter, testImages)
+        logging.info(f">> [Image Segmentation] Segmentation prediction for the batch {startIndex // batchSize + 1}/{dataset.count // batchSize + 1} is complete.")
+        batchPrediction = [np.array(element) for element in prediction.tolist()]
+
+        predictedMasks = addPadding(batchPrediction, testMasksPadding)
+
+        batchIouScore = iouScoreImages(testMasks, predictedMasks, batchSamples)
+        iouScores.extend(batchIouScore)
+
+        for testImg, testSeg, predSeg, iou, sample in zip(testImages, testMasks, predictedMasks, batchIouScore, batchSamples):
+            plotImageSegmentation(testImg, testSeg, predSeg, round(iou, 2), str(sample.id), taskRun)
+            logging.info(f">> [Image Segmentation] The results image for sample {sample.name} (sample id: {sample.id}) add to the artifacts.")
+            csvSamplesData.append(dict(zip(fieldNamesSamples, [str(sample.id), sample.name, str(round(iou, 2)), str(round(iou * 100))])))
+
+    sampleResultsPath = f"{folder_manager.temp}/sample_results.csv"
```

Comment:
Let's use `pathlib` for handling paths instead of treating them as strings. `folder_manager.temp` is already an instance of `pathlib.Path` class.